### PR TITLE
feat: added avatar tailoring and uploading

### PR DESCRIPTION
**Pull request by Kininaru**
Add: https://github.com/casbin/casdoor/issues/18

### CODE REVIEW

File: object/user.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type User struct {
 	Password      string `xorm:"varchar(100)" json:"password"`
 	PasswordType  string `xorm:"varchar(100)" json:"passwordType"`
 	DisplayName   string `xorm:"varchar(100)" json:"displayName"`
-	Avatar        string `xorm:"varchar(100)" json:"avatar"`
+	Avatar        string `xorm:"varchar(1000)" json:"avatar"`
```
Comment (hsluoyz) [inline object/user.go:33]: Why, 100 isn't enough for image link?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/user.go (hunk 1 above)
Comment (Kininaru) [reply object/user.go:33]: 100 is not enough for image link. For example, casdoor deployed at https://door-hk.chromium.link, has an avatar link: `https://aliyun-wb-bl6qnlpj4m.oss-cn-shanghai.aliyuncs.com/casdoor/avatar/admin.png?time=1615906437522268542`, it has 107 characters.

File: web/package.json [hunk 2]
```diff
@@ -11,12 +11,14 @@
...
     "react": "^16.14.0",
+    "react-cropper": "^2.1.7",
     "react-device-detect": "^1.14.0",
     "react-dom": "^16.14.0",
     "react-github-corner": "^2.5.0",
     "react-i18next": "^11.8.7",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "3.4.3"
+    "react-scripts": "3.4.3",
+    "typescript": "^4.2.3"
```
[the remark this answers is not in the document]
Comment (Kininaru) [reply web/package.json]: Removed ts

**Review by hsluoyz — CHANGES_REQUESTED** [review]
Why isn't yarn.lock updated?

**Comment by Kininaru** [comment]
Updated yarn.lock now. @hsluoyz

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**